nucypher/datastore/datastore.py

```diff
@@ -180,7 +125,7 @@ def attach_kfrag_to_saved_arrangement(self, alice, id_as_hex, kfrag, session=Non
         if policy_arrangement is None:
             raise NotFound("Can't attach a kfrag to non-existent Arrangement {}".format(id_as_hex))
 
-        if policy_arrangement.alice_verifying_key.key_data != alice.stamp:
+        if policy_arrangement.alice_verifying_key != alice.stamp:
             raise alice.SuspiciousActivity
 
         policy_arrangement.kfrag = bytes(kfrag)
```

Comment:
This record is not persisted correctly, see #2088.

Reply:
Can we get a formal "production" test for this before merging this changeset?

Reply:
Yes - let's do an experiment on testnet sooner vs. later!

Reply:
Before this is merged, preferably. If there's an issue, I need to know _now_.

Reply:
After the recent tests, I'm going to consider this resolved now.